addons/web_editor/static/src/js/editor/odoo-editor/src/utils/utils.js

```diff
@@ -456,6 +456,19 @@ export function getAdjacents(node, predicate = n => !!n) {
 export function hasTableSelection(editable) {
     return !!editable.querySelector('.o_selected_table');
 }
+/**
+ * Checks if the entire table associated with the given range is fully selected.
+ *
+ * @param {Range} range
+ * @returns {boolean} - True if the entire table is fully selected, false otherwise.
+ */
+export function hasTableFullySelected(range) {
```

Comment:
Please rename to `isTableFullySelected`. Also, would this work if the range is not deep? If not, then your function should start by calling `getDeepRange` to replace the range passed as argument.